src/main/java/com/google/sps/servlets/GmailServlet.java

```diff
@@ -66,15 +74,131 @@ public void doGet(
     assert googleCredential != null
         : "Null credentials (i.e. unauthenticated requests) should already be handled";
 
-    // Get messageIds from Gmail
     GmailClient gmailClient = gmailClientFactory.getGmailClient(googleCredential);
-    List<Message> messages = gmailClient.listUserMessages("");
 
-    // convert messageIds to JSON object and print to response
-    Gson gson = new Gson();
-    String messageJson = gson.toJson(messages);
+    int nDays;
+    int mHours;
+    try {
+      nDays = Integer.parseInt(request.getParameter("nDays"));
+      mHours = Integer.parseInt(request.getParameter("mHours"));
+    } catch (NumberFormatException e) {
+      response.sendError(400, "nDays and mHours must be integers");
+      return;
+    }
+
+    if (nDays < 0 || mHours < 0) {
+      response.sendError(400, "nDays and mHours must be positive");
+      return;
+    }
 
+    // TODO: Perform these operations faster (use multithreading, batching, etc) (Issue #84)
+    int unreadEmails = getUnreadEmailCountDays(nDays, gmailClient);
+    int recentUnreadEmails = getUnreadEmailCountHours(mHours, gmailClient);
+    int unreadImportantEmails = getImportantUnreadEmailCount(nDays, gmailClient);
+    String sender = getMostFrequentSender(nDays, gmailClient);
+    GmailResponse gmailResponse =
+        new GmailResponse(
+            nDays, mHours, unreadEmails, recentUnreadEmails, unreadImportantEmails, sender);
+
+    Gson gson = new Gson();
+    String messageJson = gson.toJson(gmailResponse);
     response.setContentType("application/json");
     response.getWriter().println(messageJson);
   }
+
+  /**
+   * Get the amount of unread emails from the last n days in a user's Gmail account
+   *
+   * @param nDays number of days to check the user's account for
+   * @param gmailClient Gmail service with valid google credential
+   * @return number of unread emails from last nDays days
+   * @throws IOException if an issue occurs with the Gmail Service
+   */
+  private int getUnreadEmailCountDays(int nDays, GmailClient gmailClient) throws IOException {
+    String searchQuery = GmailClient.emailQueryString(nDays, "d", true, false, "");
```

Comment:
Might be worth explaining what `"d"`, `true`, `false` and `""` represent in `.emailQueryString` or assigning them to constants like you did in GmailClientTest.

Reply:
This is documented in the `GmailClient.emailQueryString()` javadoc. I think it might be redundant to repeat this here.